src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1519,4 +1519,13 @@
     <value>and all other platforms</value>
     <comment>This call site is reachable on: 'windows' 10.0.2000 and later, and all other platforms</comment>
   </data>
+  <data name="UseAssignableTypeForForeachVariableDescription" xml:space="preserve">
+    <value>There is no implicit conversion between collection element and iteratoi variable, this will result in runtime exception.</value>
+  </data>
+  <data name="UseAssignableTypeForForeachVariableMessage" xml:space="preserve">
+    <value>Cannot convert type '{0}' to '{1}'</value>
+  </data>
+  <data name="UseAssignableTypeForForeachVariableTitle" xml:space="preserve">
+    <value>Use valid foreach variable type</value>
```

Comment:
What about osmething like: `No implicit conversion between these types`? Or `Cannot convert these types`.

Reply:
Updated to `Cannot implicitly convert type 'type1' to 'type2'.` like in [CS0266](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0266)

Reply:
This analyzer only checks foreach. For other places there are already [CS0266](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0266) and [CS0030](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0030). Any they usually work, just not here.

Anyway, I think analyzer name should include `foreach` somewhere.

Reply:
What about `Use implicitly assignable foreach variable type`? It conveys the meaning pretty well.